File: Examen_Unidad2/src/Visita.java
import java.util.*;
class Visita {
    private static int counter = 1;
    private String id;
    private String guia;
    private List<Visitante> visitantes;
    private double costoTotal;
    private String fechaVisita;
    private int cantidadNinos;
    private int cantidadAdultos;

    public Visita(String guia, List<Visitante> visitantes, double costoTotal, String fechaVisita, int cantidadNinos, int cantidadAdultos) {
        this.id = "VIS" + counter++;
        this.guia = guia;
        this.visitantes = visitantes;
        this.costoTotal = costoTotal;
        this.fechaVisita = fechaVisita;
        this.cantidadNinos = cantidadNinos;
        this.cantidadAdultos = cantidadAdultos;
    }

    public String getId() {
        return id;
    }

    public String getGuia() {
        return guia;
    }

    public void setGuia(String guia) {
        this.guia = guia;
    }

    public List<Visitante> getVisitantes() {
        return visitantes;
    }

    public void setVisitantes(List<Visitante> visitantes) {
        this.visitantes = visitantes;
    }

    public double getCostoTotal() {
        return costoTotal;
    }

    public void setCostoTotal(double costoTotal) {
        this.costoTotal = costoTotal;
    }

    public String getFechaVisita() {
        return fechaVisita;
    }

    public void setFechaVisita(String fechaVisita) {
        this.fechaVisita = fechaVisita;
    }

    public int getCantidadNinos() {
        return cantidadNinos;
    }

    public void setCantidadNinos(int cantidadNinos) {
        this.cantidadNinos = cantidadNinos;
    }

    public int getCantidadAdultos() {
        return cantidadAdultos;
    }

    public void setCantidadAdultos(int cantidadAdultos) {
        this.cantidadAdultos = cantidadAdultos;
    }
    public String toString() {
        return "Visita" +
                "Id: '" + id + '\'' +
                ", Guia: '" + guia + '\'' +
                ", Visitantes: " + visitantes +
                ", Costo Total: " + costoTotal +
                ", Fecha Visita. '" + fechaVisita + '\'' +
                ", antidad Ninos: " + cantidadNinos +
                ", Cantidad Adultos: " + cantidadAdultos;
    }
}

File: Examen_Unidad2py/ZOO PY/Visita.py
class Visita:
  counter = 1

  def __init__(self, guia, visitantes, costoTotal, fechaVisita, cantidadNinos, cantidadAdultos):
      self.id = "VIS" + str(Visita.counter)
      Visita.counter += 1
      self.guia = guia
      self.visitantes = visitantes
      self.costoTotal = costoTotal
      self.fechaVisita = fechaVisita
      self.cantidadNinos = cantidadNinos
      self.cantidadAdultos = cantidadAdultos

  def getId(self):
      return self.id

  def getGuia(self):
      return self.guia

  def setGuia(self, guia):
      self.guia = guia

  def getVisitantes(self):
      return self.visitantes

  def setVisitantes(self, visitantes):
      self.visitantes = visitantes

  def getCostoTotal(self):
      return self.costoTotal

  def setCostoTotal(self, costoTotal):
      self.costoTotal = costoTotal

  def getFechaVisita(self):
      return self.fechaVisita

  def setFechaVisita(self, fechaVisita):
      self.fechaVisita = fechaVisita

  def getCantidadNinos(self):
      return self.cantidadNinos

  def setCantidadNinos(self, cantidadNinos):
      self.cantidadNinos = cantidadNinos

  def getCantidadAdultos(self):
      return self.cantidadAdultos

  def setCantidadAdultos(self, cantidadAdultos):
      self.cantidadAdultos = cantidadAdultos

  def __str__(self):
      return f"Visita: Id: '{self.id}', Guia: '{self.guia}', Visitantes: {self.visitantes}, Costo Total: {self.costoTotal}, Fecha Visita: '{self.fechaVisita}', Cantidad Ninos: {self.cantidadNinos}, Cantidad Adultos: {self.cantidadAdultos}"
